test/remote/sms_tests.js

```diff
@@ -38,3 +39,73 @@ describe('remote sms', () => {
     return TestServer.stop(server)
   })
 })
+
+describe('remote sms (mocked nexmo)', () => {
+  let server
+
+  before(() => {
+    config.sms.enabled = true
+    // POST /sms spends actual money unless the SMS provider is mocked
+    config.sms.useMock = true
+
+    return TestServer.start(config)
+      .then(result => {
+        server = result
+      })
+  })
+
+  it('POST /sms success', () => {
```

Comment:
Fwiw, I'm not sure whether I agree.

There is a remote test of `GET /sms/status` above this at line 26, which asserts that endpoint in a proper end-to-end sense. The only reason I opted to do it this way instead for `POST /sms` was because I didn't think we wanted to spend money every time somebody runs the tests, so that endpoint couldn't have an unmocked test. A proper end-to-end test is preferable, I think.

Then there are tests for `GET /status` that use mocked responses from Nexmo in `test/local/routes/sms.js`. I think those assert all of the finer-grained logic that is driven by the Nexmo response itself.

Or if you meant a test to ensure the mock itself is working, I think that's covered by `test/local/lib/mock-nexmo.js`.

So I'm not sure what assertions I could write that would add more value here. Generally I'm against adding tests that don't add value because they increase the cost of refactoring.

But also I'm happy to be persuaded to the contrary if you have some specific assertions in mind that you don't think are covered yet.